addons/sms/wizard/sms_composer.py

```diff
@@ -86,7 +52,39 @@ def default_get(self, fields):
     sanitized_numbers = fields.Char('Sanitized Number', compute='_compute_sanitized_numbers', compute_sudo=False)
     # content
     template_id = fields.Many2one('sms.template', string='Use Template', domain="[('model', '=', res_model)]")
-    body = fields.Text('Message', required=True)
+    body = fields.Text('Message', required=True, store=True, compute='_compute_body', readonly=False)
+
+    @api.depends('active_domain', 'res_id', 'res_ids')
+    def _compute_composition_mode(self):
+        for record in self:
+            record.composition_mode = record.env.context.get('default_composition_mode')
+            if record.env.context.get('active_ids') and len(record.env.context.get('active_ids')) > 1:
+                record.composition_mode = 'mass'
+                record.res_id = False
+            else:
+                record.composition_mode = 'comment'
+                record.res_ids = False
+            
+            if not record.env.context.get('active_domain'):
+                record.active_domain = repr(record.env.context.get('active_domain', []))
+            if not record.env.context.get('res_id'):
+                if not record.env.context.get('res_ids') and record.env.context.get('active_id'):
+                    record.res_id = record.env.context.get('active_id')
+            if not record.env.context.get('res_ids'):
+                if not record.env.context.get('res_id') and record.env.context.get('active_ids'):
+                    record.res_ids = repr(record.env.context.get('active_ids'))
+
+    @api.depends('composition_mode', 'res_id', 'body', 'template_id')
+    def _compute_res_model(self):
+        for record in self:
+            record.res_model = record.env.context.get('default_res_model')
+            if record.res_model:
+                record.update(
+                    record._get_composer_values(
```

Comment:
This should not be necessary as body depends on res_model / res_id / template_id / composition_mode. When creating the composer it will trigger the field computation automatically normally. If a field is computed, we should avoid trying to compute it manually unless some specific cases and let the ORM do its job.

Probably that _get_composer_values will become useless with this PR by the way.